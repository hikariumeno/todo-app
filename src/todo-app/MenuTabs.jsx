import React from "react";
import "bootstrap/dist/css/bootstrap.min.css";
import Tabs from "react-bootstrap/Tabs";
import Tab from "react-bootstrap/Tab";

export class MenuTabs extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      activeTab: "All",
    };
    this.handleSelect = this.handleSelect.bind(this);
  }

  handleSelect(event) {
    this.setState({ activeTab: event });
  }

  render() {
    return (
      <Tabs fill activeKey={this.state.activeTab} onSelect={this.handleSelect}>
        <Tab eventKey="All" title="All"></Tab>
        <Tab eventKey="Active" title="Active"></Tab>
        <Tab eventKey="Completed" title="Completed"></Tab>
      </Tabs>
    );
  }
}
