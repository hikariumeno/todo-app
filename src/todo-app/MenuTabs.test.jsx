import React from "react";
import renderer from "react-test-renderer";
import { MenuTabs } from "./MenuTabs";

test("All Tab selected", () => {
  const component = renderer.create(
    <MenuTabs activeKey="All" onSelect="Function"></MenuTabs>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
