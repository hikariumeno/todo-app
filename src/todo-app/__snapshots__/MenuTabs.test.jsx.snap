// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActiveTab changes the class selected 1`] = `
Array [
  <nav
    className="nav nav-tabs nav-fill"
    onKeyDown={[Function]}
    role="tablist"
  >
    <a
      aria-selected={true}
      className="nav-item nav-link active"
      data-rb-event-key="All"
      href="#"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="tab"
    >
      All
    </a>
    <a
      aria-selected={false}
      className="nav-item nav-link"
      data-rb-event-key="Active"
      href="#"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="tab"
    >
      Active
    </a>
    <a
      aria-selected={false}
      className="nav-item nav-link"
      data-rb-event-key="Completed"
      href="#"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="tab"
    >
      Completed
    </a>
  </nav>,
  <div
    className="tab-content"
  >
    <div
      aria-hidden={false}
      aria-labelledby={null}
      className="fade tab-pane active show"
      id={null}
      role="tabpanel"
    />
    <div
      aria-hidden={true}
      aria-labelledby={null}
      className="fade tab-pane"
      id={null}
      role="tabpanel"
    />
    <div
      aria-hidden={true}
      aria-labelledby={null}
      className="fade tab-pane"
      id={null}
      role="tabpanel"
    />
  </div>,
]
`;

exports[`All Tab selected 1`] = `
Array [
  <nav
    className="nav nav-tabs nav-fill"
    onKeyDown={[Function]}
    role="tablist"
  >
    <a
      aria-selected={true}
      className="nav-item nav-link active"
      data-rb-event-key="All"
      href="#"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="tab"
    >
      All
    </a>
    <a
      aria-selected={false}
      className="nav-item nav-link"
      data-rb-event-key="Active"
      href="#"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="tab"
    >
      Active
    </a>
    <a
      aria-selected={false}
      className="nav-item nav-link"
      data-rb-event-key="Completed"
      href="#"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="tab"
    >
      Completed
    </a>
  </nav>,
  <div
    className="tab-content"
  >
    <div
      aria-hidden={false}
      aria-labelledby={null}
      className="fade tab-pane active show"
      id={null}
      role="tabpanel"
    />
    <div
      aria-hidden={true}
      aria-labelledby={null}
      className="fade tab-pane"
      id={null}
      role="tabpanel"
    />
    <div
      aria-hidden={true}
      aria-labelledby={null}
      className="fade tab-pane"
      id={null}
      role="tabpanel"
    />
  </div>,
]
`;
